src/main/java/org/opensrp/domain/IdentifierSource.java

```diff
@@ -0,0 +1,151 @@
+package org.opensrp.domain;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.opensrp.util.IdentifierValidatorAlgorithm;
+
+import javax.persistence.EnumType;
+import javax.persistence.Enumerated;
+
+public class IdentifierSource {
+
+	@JsonProperty
+	private Long id;
+
+	@JsonProperty
+	private String identifier;
+
+	@JsonProperty
+	private String description;
+
+	@Enumerated(EnumType.STRING)
+	@JsonProperty
+	private IdentifierValidatorAlgorithm identifierValidatorAlgorithm;
+
+	@JsonProperty
+	private String baseCharacterSet;
+
+	@JsonProperty
+	private String firstIdentifierBase;
+
+	@JsonProperty
+	private String prefix;
+
+	@JsonProperty
+	private String suffix;
+
+	@JsonProperty
+	private Integer minLength;
+
+	@JsonProperty
+	private Integer maxLength;
+
+	@JsonProperty
+	private String blacklisted;
```

Comment:
I think this should be part of the **core.unique_Id**'s schema since we wan't to black list numbers that have already been generated from a specific source

Reply:
To keep a track of reserved identifiers, I believe a field like reserved_identifiers would be added in core.unique_ids. The purpose of blacklisted is to identify the numbers that should never be generated and it would be a Regex. @ndegwamartin 
Reference # https://github.com/OpenSRP/opensrp-server-core/issues/197